engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpression.scala

```diff
@@ -404,10 +405,12 @@ object SpelExpressionParser extends LazyLogging {
 
     protected def findMethod(name: String, target: Any): Option[Method] = {
       val targetClass = extractClassFromTarget(target)
+        //TODO: better exception message
```

Comment:
Logic of Caching PropertyAccessors is complicated. In concrete class, in `canRead` we invoke `findMethod` from (two level up) `CachingBase` which invokes `extractClassFromTarget` (from one level down) `Caching` which check some nulls. Maybe this is the moment to simplify it somehow? I wonder if we can have only one level of inheritance and null check at the beginning (`canRead`) to avoid checking it in nested invocations? Maybe some separate `PropertyAccessor` for nulls? Or maybe some composition will help (like extracting new `ClassMethodsMemo` class)? For now I'm not sure why you changed it that way and what exactly is fixed...

Reply:
I don't fix too much here :) 
I just don't want to throw "raw" NullpointerException, as it is later reported as "Unknown Exception", whereas I think it's kind of "normal" that user created diagram where she invokes method on null - it's an exception, but nothing unexpected... 

In https://travis-ci.org/TouK/nussknacker/jobs/613989223
org.springframework.expression.spel.SpelEvaluationException: EL1021E: A problem occurred whilst attempting to access the property 'name': 'null'
...
Caused by: java.lang.NullPointerException: null
	at pl.[secure].nussknacker.engine.spel.SpelExpressionParser$ScalaLazyPropertyAccessor.reallyFindMethod(SpelExpression.scala:307) ~[classes/:na]

And in https://travis-ci.org/TouK/nussknacker/jobs/614155476:

18:56:58.476 [pool-1-thread-1-ScalaTest-running-FlinkTestMainSpec] INFO  p.t.n.engine.spel.SpelExpression - Expression evaluation failed. Original #input.value1.toString(), ctxId: , message: EL1021E: A problem occurred whilst attempting to access the property 'value1': 'Cannot invoke method/property value1 on null object'

Reply:
Ok, good logging cleanup. WDYT about extracting NullPropertyAccessor and in other PropertyAccessors we won't worry about it?